### PR TITLE
Add missing rules and make names nicer

### PR DESCRIPTION
**Pull request by jaredfholgate**
- Use reflection to add WAF rules dynamically
- Make the name nicer by removing the prefix
- Add the ability to override a rule name in case of a collision

### CODE REVIEW

**Comment by jaredfholgate** [comment]
@matt-FFFFFF I overlooked adding the rules to the collection. I figured it was worth making it slightly simpler while I was in here to avoid other people forgetting. There is not a need to add the methods to the struct, but hopefully simpler than remembering to add them in the rules var?

File: waf/waf.go [hunk 1]
```diff
@@ -3,21 +3,24 @@
...
-	AzurermPublicIpSku(),
-	AzurermPublicIpZones(),
-	AzurermServicePlanZoneBalancingEnabled(),
-	AzurermStorageAccountAccountReplicationType(),
-	AzurermVirtualMachineZoneUnknown(),
-	AzurermVirtualMachineZonesUnknown(),
-	AzurermVirtualNetworkGatewaySku(),
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
```
Comment (matt-FFFFFF) [inline waf/waf.go:11]: Sorry to nit pick but can we add some docs to this type, and to the `GetRules` func below?


```suggestion
// WafRules is a helper struct. Methods are created on this type that generate the rules for the WAF package.
// We then use reflection to automatically generate a slice of the rules to add the the ruleset.
// This avoids having to manually maintain a list of rules.
// See `GetRules()` for more detail.
```

File: waf/waf.go [hunk 2]
```diff
@@ -3,21 +3,27 @@
...
-	AzurermServicePlanZoneBalancingEnabled(),
-	AzurermStorageAccountAccountReplicationType(),
-	AzurermVirtualMachineZoneUnknown(),
-	AzurermVirtualMachineZonesUnknown(),
-	AzurermVirtualNetworkGatewaySku(),
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+// WafRules is a helper struct. Methods are created on this type that generate the rules for the WAF package.
+// We then use reflection to automatically generate a slice of the rules to add the the ruleset.
+// This avoids having to manually maintain a list of rules.
```
Comment (matt-FFFFFF) [inline waf/waf.go:13]: as above


```suggestion

GetRules uses reflection to iterate over all the methods of the WafRules struct and add them to a slice of Rules to be included in the ruleset.
```

**Review by matt-FFFFFF — APPROVED** [review]

File: waf/waf.go [hunk 3]
```diff
@@ -3,21 +3,24 @@
...
+func GetRules() []tflint.Rule {
+	rules := []tflint.Rule{}
+
+	wafRules := reflect.TypeOf(WafRules{})
+
+	for i := 0; i < wafRules.NumMethod(); i++ {
+		method := wafRules.Method(i)
+		rule := reflect.ValueOf(WafRules{}).MethodByName(method.Name).Call([]reflect.Value{})
+		rules = append(rules, rule[0].Interface().(tflint.Rule))
+	}
```
[the remark this answers is not in the document]
Comment (jaredfholgate) [reply waf/waf.go]: I made further updates to make this a lot more readable and added inline comments given it is less than obvious if not familiar with reflection.

**Pull request merged**